driver-core/build.gradle

```diff
@@ -31,17 +31,17 @@ ext {
 sourceSets.main.resources.srcDirs = ['src/resources']
 
 dependencies {
-    compile project(path: ':bson', configuration: 'default')
+    api project(path: ':bson', configuration: 'default')
 
-    compile "com.github.jnr:jnr-unixsocket:$jnrUnixsocketVersion", optional
-    compile "io.netty:netty-buffer:$nettyVersion", optional
-    compile "io.netty:netty-transport:$nettyVersion", optional
-    compile "io.netty:netty-handler:$nettyVersion", optional
-    compile "org.xerial.snappy:snappy-java:$snappyVersion", optional
-    compile "com.github.luben:zstd-jni:$zstdVersion", optional
-    compile "org.mongodb:mongodb-crypt:$mongoCryptVersion", optional
+    api "com.github.jnr:jnr-unixsocket:$jnrUnixsocketVersion", optional
+    api "io.netty:netty-buffer:$nettyVersion", optional
+    api "io.netty:netty-transport:$nettyVersion", optional
+    api "io.netty:netty-handler:$nettyVersion", optional
+    api "org.xerial.snappy:snappy-java:$snappyVersion", optional
```

Comment:
I don't think any of these three need to be API.

Reply:
Changing these from `api` to `implementation` will change the scope in the POM from `compile` to `runtime`. If that is an acceptable change, I will make it so.

Reply:
```
    <dependency>
      <groupId>org.xerial.snappy</groupId>
      <artifactId>snappy-java</artifactId>
      <version>1.1.4</version>
      <scope>runtime</scope>
      <optional>true</optional>
    </dependency>
    <dependency>
      <groupId>com.github.luben</groupId>
      <artifactId>zstd-jni</artifactId>
      <version>1.3.8-3</version>
      <scope>runtime</scope>
      <optional>true</optional>
    </dependency>
    <dependency>
      <groupId>org.mongodb</groupId>
      <artifactId>mongodb-crypt</artifactId>
      <version>1.0.0</version>
      <scope>runtime</scope>
      <optional>true</optional>
    </dependency>
```

Reply:
Yes, this is what we want.  Though arguably it doesn't matter all that match since the dependencies are optional anyway.

Reply:
Good. I have switched back to `implementation` for the library dependencies.